android/KMEA/app/src/main/java/com/keyman/engine/cloud/impl/CloudLexicalModelMetaDataDownloadCallback.java

```diff
@@ -153,19 +153,21 @@ private void processCloudResultForModel(Context aContext, MetaDataResult _r) {
     }
 
     try {
-      JSONObject modelInfo = lmData.getJSONObject(0);
-      if (!modelInfo.has("packageFilename") || !modelInfo.has("id")) {
-        KMLog.LogError(TAG, "Error in lexical model metadata from api.keyman.com - missing metadata");
-        return;
-      }
-
-      String _modelID = modelInfo.getString("id");
       ArrayList<CloudApiTypes.CloudApiParam> urls = new ArrayList<>();
-      urls.add(new CloudApiTypes.CloudApiParam(
-        CloudApiTypes.ApiTarget.LexicalModelPackage,
-        modelInfo.getString("packageFilename")));
-      _r.additionalDownloadid = CloudLexicalPackageDownloadCallback.createDownloadId(_modelID);
-      _r.additionalDownloads= urls;
+      for(int i=0; i< lmData.length(); i++) {
+        JSONObject modelInfo = lmData.getJSONObject(i);
+        if (!modelInfo.has("packageFilename") || !modelInfo.has("id")) {
+          KMLog.LogError(TAG, "Error in lexical model metadata from api.keyman.com - missing metadata");
+          return;
```

Comment:
```suggestion
          continue;
```

Might as well try and install another one?